pilot/pkg/networking/util/util.go

```diff
@@ -523,6 +522,31 @@ func BuildLbEndpointMetadata(network string, tlsMode string) *core.Metadata {
 		}
 	}
 
+	// Add compressed telemetry metadata. Note this is a short term solution to make server workload metadata
+	// available at client sidecar, so that telemetry filter could use for metric labels. This is useful for two cases:
+	// server does not have sidecar injected, and request fails to reach server and thus metadata exchange does not happen.
+	// Due to performance concern, telemetry metadata is compressed into a semicolon separted string:
+	// workload-name;namespace;canonical-service-name;canonical-service-revision.
+	if features.EndpointTelemetryLabel {
+		var sb strings.Builder
+		sb.WriteString(name)
+		sb.WriteString(";")
+		sb.WriteString(namespace)
+		sb.WriteString(";")
+		if csn, ok := labels[model.IstioCanonicalServiceLabelName]; ok {
+			sb.WriteString(csn)
+		}
+		sb.WriteString(";")
+		if csr, ok := labels[model.IstioCanonicalServiceRevisionLabelName]; ok {
+			sb.WriteString(csr)
+		}
+		metadata.FilterMetadata[IstioMetadataKey] = &pstruct.Struct{
```

Comment:
won't this collide with the `network` label? Both are doing `metadata.FilterMetadata[IstioMetadataKey] = &pstruct.Struct ...`

Reply:
Rather than adding a new value into `metadata.FilterMetadata[IstioMetadataKey].Fields`

Reply:
Fixed. Thanks for catching this! I reversed the verification condition in the test so missed this bug there originally :p.